eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/Main.java

```diff
@@ -30,35 +32,48 @@ private static BomGenerator parseCommandLine(String[] args) {
 
                     switch (argName.toLowerCase()) {
```

Comment:
If any of the arguments are missing will the application fail to run? If so, I'd revert back to having all parameters being required in the constructor.

Reply:
No, mode is an optional parameter. And in the next iteration, I intend to add another optional one.